auto_nag/scripts/inactive_ni_pending.py

```diff
@@ -0,0 +1,184 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from dateutil.relativedelta import relativedelta
+from libmozdata import utils as lmdutils
+
+from auto_nag import utils
+from auto_nag.bzcleaner import BzCleaner
+from auto_nag.user_activity import UserActivity
+
+# TODO: should be moved when resolving https://github.com/mozilla/relman-auto-nag/issues/1384
+HIGH_PRIORITY = {"P1", "P2"}
+HIGH_SEVERITY = {"S1", "critical", "S2", "major"}
+
+
+class InactiveNeedinfoPending(BzCleaner):
+    def __init__(self):
+        super(InactiveNeedinfoPending, self).__init__()
+        self.max_actions = utils.get_config(self.name(), "max_actions", 7)
+        self.__recent_bug_limit = lmdutils.get_date_ymd("today") - relativedelta(
+            months=6
+        )
+
+    def get_max_actions(self):
+        return self.max_actions
+
+    def description(self):
+        return "Bugs with needinfo pending on inactive people"
+
+    def columns(self):
+        return ["id", "summary", "inactive_ni", "inactive_ni_count"]
+
+    def get_bugs(self, *args, **kwargs):
+        bugs = super().get_bugs(*args, **kwargs)
+        self.handle_inactive_requestee(bugs)
+
+        # Resolving https://github.com/mozilla/relman-auto-nag/issues/1300 should clean this
+        # including improve the wording in the template (i.e., "See the search query on Bugzilla").
+        self.query_url = utils.get_bz_search_url({"bug_id": ",".join(bugs.keys())})
+
+        return bugs
+
+    def handle_inactive_requestee(self, bugs):
+        requestee_bugs = {}
+        for bugid, bug in bugs.items():
+            for flag in bug["needinfo_flags"]:
+                if "requestee" not in flag:
+                    flag["requestee"] = ""
+
+                if flag["requestee"] not in requestee_bugs:
+                    requestee_bugs[flag["requestee"]] = [bugid]
+                else:
+                    requestee_bugs[flag["requestee"]].append(bugid)
+
+        user_activity = UserActivity()
+        inactive_users = user_activity.check_users(requestee_bugs.keys())
+        selected_bugs = []
+        for requestee, bugids in requestee_bugs.items():
+            if requestee in inactive_users:
+                selected_bugs.extend(bugids)
+
+        for bugid in set(bugs.keys()) - set(selected_bugs):
+            del bugs[bugid]
```

Comment:
Nit: you could use a dict comprehension:
```Python
bugs = {bugid: bug for bugid, bug in bugs.items() if bugid in set(selected_bugs)}
```

Reply:
I was preferring the current to behaviour since it makes it clear that we are modifying the provided `bugs` object (since we are not returning a new one). Using the dict comprehension will have it less explicit since we are returning a new and modifying bug object in the provided one. In d35ea849ee0082b4beb60681a456ac9ef7f12104. I switched to using dict comprehension as you suggested. I mitigate the clarity issue by adding a clarification comment.